### PR TITLE
docs(card): update Style subpage

### PR DESCRIPTION
**Pull request by marionnegp**
## What I did

Made the following updates to card's [Theme section](https://deploy-preview-1664--red-hat-design-system.netlify.app/elements/card/style/#theme):
- Added text for the Theme section
- Added headings for light and dark theme sections
- Updated images for light and dark themes
- Added copy for Color section
- Changed Color examples from images to `<rh-card>` elements

## Testing Instructions

Check the [card's Theme section](https://deploy-preview-1664--red-hat-design-system.netlify.app/elements/card/style/#theme) in the DP.

## Notes to Reviewers
It looks like the site's style.css file is adding a margin above all `<p>` tags. This is a known issue with `<rh-card>` and should be tackled in a separate issue/PR.

Screenshot 2024-06-26 at 2 57 39 PM

### CODE REVIEW

File: elements/rh-card/docs/10-style.md [hunk 1]
```diff
@@ -23,59 +23,63 @@ elements and styles can be placed inside.
...
+    <rh-card color-palette="dark">
+      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam eleifend elit sed est egestas, a sollicitudin mauris tincidunt.</p>
+      <rh-cta href="#">Call to action</rh-cta>
+    </rh-card>
     <figcaption>
-      A black card with a dark gray border is the most common use case in the
-      dark theme
+      A black card with a dark gray border is the most common use case in the dark theme.
     </figcaption>
   </figure>
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:71]: I wonder if for the dark cards here if we dont want to keep the `<uxdot-example color-palette="dark">` and place the card inside of it without setting its `color-palette`:

```Suggestion
<figure>
    <uxdot-example no-border color-palette="dark">
 <rh-card>
      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam eleifend elit sed est egestas, a sollicitudin mauris tincidunt.</p>
      <rh-cta href="#">Call to action</rh-cta>
    </rh-card>
</uxdot-example>
    <figcaption>
      A black card with a dark gray border is the most common use case in the dark theme.
    </figcaption>
  </figure>
```

Screenshot of what this renders as:

Screenshot 2024-06-27 at 3 00 09 PM

It's my thought that this more accurately represents the figcaption showing the grey border on a dark theme.

File: elements/rh-card/docs/10-style.md [hunk 2]
```diff
@@ -23,59 +23,63 @@ elements and styles can be placed inside.
...
+
   <figure>
-    <uxdot-example color-palette="darkest" width-adjustment="242px">
-      <img src="{{ '../card-color-dark-gray.svg' | url }}" alt="Dark theme gray card colors">
-    </uxdot-example>
+    <rh-card color-palette="darker">
```
[the remark this answers is not in the document]
Comment (zeroedin) [reply elements/rh-card/docs/10-style.md:74]: That said changing the value seems to work so might be a JSDoc documentation issue. 

@bennypowers thoughts?

[the remark this answers is not in the document]
Comment (marionnegp) [reply elements/rh-card/docs/10-style.md]: Good suggestion

File: elements/rh-card/docs/10-style.md (hunk 1 above)
Comment (marionnegp) [reply elements/rh-card/docs/10-style.md]: Good suggestion

File: elements/rh-card/docs/10-style.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (marionnegp) [reply elements/rh-card/docs/10-style.md]: I don't really see why you wouldn't be able to use all of the surface colors in card. @coreyvickery, any reasons why not? We would have to add these color options to the library.

[the remark this answers is not in the document]
Comment (bennypowers) [reply elements/rh-card/docs/10-style.md]: which color palettes are supported by card it out of scope of this pr

**Pull request merged**